save-backend/src/main/kotlin/org/cqfn/save/backend/configs/WebConfiguration.kt

```diff
@@ -28,8 +28,14 @@ class WebConfiguration {
      */
     @Bean
     fun indexRouter(@Value("classpath:/static/index.html") html: Resource) = router {
-        GET("/") {
-            ok().header("Content-Type", "text/html; charset=utf8").bodyValue(html)
+        GET("/") { request ->
+            request.principal().flatMap {
+                // if user is logged in, show `index.html`
+                ok().header("Content-Type", "text/html; charset=utf8").bodyValue(html)
+            }.switchIfEmpty {
+                // if not, redirect to the login page
+                status(HttpStatus.FOUND).header("Location", "/login").build()
```

Comment:
we have a beatiful NOT_FOUND page, will it be in place?

Reply:
We now only display it from react app, we should make it default fallback page under #409